nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -811,6 +835,24 @@ in {
 
       environment.systemPackages = [ occ ];
 
+      services.mysql = lib.mkIf cfg.database.createLocally {
+        enable = true;
+        package = lib.mkDefault pkgs.mariadb;
+        ensureDatabases = [ cfg.config.dbname ];
+        ensureUsers = [{
+          name = cfg.config.dbuser;
+          ensurePermissions = { "${cfg.config.dbname}.*" = "ALL PRIVILEGES"; };
+        }];
+        # FIXME(@Ma27) Nextcloud isn't compatible with mariadb 10.6,
+        # this is a workaround.
+        # See https://help.nextcloud.com/t/update-to-next-cloud-21-0-2-has-get-an-error/117028/22
+        settings = {
+          mysqld = {
+            innodb_read_only_compressed = 0;
```

Comment:
Not sure if we actually want to put this hack into the module itself. Not a MySQL/Mariadb user though.

cc @aanderse for opinions.

Reply:
You *could* mention in the `description` of `createLocally` that we modify the database configuration as required. Additionally we can check the type and version of `config.services.mysql.package` so we only apply this setting if needed.